### PR TITLE
Add XML stylesheet example with XSLT

### PR DESCRIPTION
**Pull request by jhabdas**
Just an example for #118.

### CODE REVIEW

File: lib/feed.xml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
+<?xml-stylesheet type="text/xml" href="feed.xslt.xml"?>
```
Comment (benbalter) [inline lib/feed.xml]: This should be `{{ url_base}}/feed.xslt.xml`

File: lib/feed.xml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
+<?xml-stylesheet type="text/xml" href="{{ url_base }}/feed.xslt.xml"?>
```
Comment (jhabdas) [inline lib/feed.xml]: Updated to use `url_base` liquid object.

**Comment by jhabdas** [comment]
Working on some basic customizations to make the default XSLT template look a little nicer and be more usable.

**Comment by jhabdas** [comment]
Made a small adjustment to the XSLT stylesheet so it outputs both the published and updated dates, and fixed a unit test for `url_base` which I may have missed an edge case for.

Working with the XSLT is tricky so I opted not to go too much further. Ideally some basic date parsing via use of [`format-dateTime`](https://www.w3.org/TR/xpath-functions-30/#func-format-dateTime) could be helpful for displaying dates. If someone wanted to go all out they could adapt [the code](https://plugins.svn.wordpress.org/xslt/trunk/) for [Better RSS Feeds](https://wordpress.org/plugins/xslt/screenshots/) to output a fully styled HTML document with hyperlinks and social sharing. For my purposes I'm happy with simply not seeing unformatted XML.

**Comment by benbalter** [comment]
> Made a small adjustment to the XSLT stylesheet so it outputs both the published and updated dates, and fixed a unit test for url_base which I may have missed an edge case for.

Would you mind posting a screenshot?

Also /cc @jglovier famed Jekyll designer, in case he has any suggestions.

File: lib/jekyll/jekyll-feed.rb [hunk 3]
```diff
@@ -47,5 +50,30 @@ def feed_exists?
...
+      xslt = PageWithoutAFile.new(@site, File.dirname(__FILE__), "", xslt_path)
+      xslt.content = File.read(xslt_source_path).gsub(/(?<!\")\s+([<{])/, '\1')
+      xslt.data["layout"] = nil
+      xslt.data["sitemap"] = false
+      xslt.output
+      xslt
+    end
+
+    # Checks if a feed stylesheet exists in the site source
+    def xslt_exists?
```
Comment (benbalter) [inline lib/jekyll/jekyll-feed.rb]: Should this an the `feed_exists?` methods be abstracted into a `file_exists?(file)` method could be shared for both?

File: lib/jekyll/jekyll-feed.rb [hunk 4]
```diff
@@ -47,5 +50,30 @@ def feed_exists?
         File.exist? Jekyll.sanitized_path(@site.source, path)
       end
     end
+
+    # Path to the feed.xslt.xml template file
+    def xslt_source_path
+      File.expand_path "../feed.xslt.xml", File.dirname(__FILE__)
+    end
+
+    def xslt_content
```
Comment (benbalter) [inline lib/jekyll/jekyll-feed.rb]: Same with `content_for_file(file)` or similar. Just trying to DRY things up where we can.

File: spec/jekyll-feed_spec.rb [hunk 5]
```diff
@@ -249,6 +249,12 @@
     end
   end
 
+  context "feed stylesheet" do
```
Comment (benbalter) [inline spec/jekyll-feed_spec.rb]: We'll also want to test that the resulting site includes the stylesheet file.

**Comment by jhabdas** [comment]
Here's a screenshot for the basic example provided. It made it immediately clear I wasn't outputting my RSS updated/modified dates properly from my Jekyll site:

screen shot 2016-06-13 at 2 46 04 pm.

I'll take a stab at those refactorings.

File: spec/jekyll-feed_spec.rb (hunk 5 above)
Comment (jhabdas) [inline spec/jekyll-feed_spec.rb]: Could you provide a little more direction on this one? I believe I was able to tackle the other two refactorings.

Comment (jhabdas) [inline spec/jekyll-feed_spec.rb]: Disregard this request. I think I got it.

**Comment by jhabdas** [comment]
@benbalter Ready for another review pass.

**Comment by benbalter** [comment]
@pathawks @parkr any strong feelings here?

**Comment by pathawks** [comment]
LGTM :shipit:

**Comment by parkr** [comment]
Fine with me!

**Comment by benbalter** [comment]
@jekyllbot: merge

**Comment by benbalter** [comment]
@jhabdas Thanks for this!

**Comment by jhabdas** [comment]
My pleasure. This was my second time writing teh Ruby. Thanks for making it fun @benbalter. This is almost better than humans.txt!
